4.analyze-components/scripts/nbconverted/1.visualize-reconstruction.r

```diff
@@ -46,63 +47,63 @@ readr::write_tsv(tcga_recon_cost_df, path = recon_file)
 
 base_dir <- file.path("figures", "TCGA")
 
-g <- plot_reconstruction_loss(tcga_recon_cost_df)
+tcga_recon_gg <- plot_reconstruction_loss(tcga_recon_cost_df)
 
 ggsave(file.path(base_dir, paste0("reconstruction_cost_", dataset, ".pdf")),
-       plot = g, dpi = 500, height = 3, width = 9)
+       plot = tcga_recon_gg, dpi = 500, height = 3, width = 9)
 ggsave(file.path(base_dir, paste0("reconstruction_cost_", dataset, ".png")),
-       plot = g, height = 3, width = 9)
+       plot = tcga_recon_gg, height = 3, width = 9)
 
-g
+tcga_recon_gg
 
 # Define the dataset to compile results for
 tcga_vae_recon_cost_df <- compile_reconstruction_data(dataset, data_focus = "vae")
 
-g <- plot_vae_training(tcga_vae_recon_cost_df)
+tcga_vae_loss_gg <- plot_vae_training(tcga_vae_recon_cost_df)
 
 ggsave(file.path(base_dir, paste0("vae_training_reconstruction_", dataset, ".pdf")),
-       plot = g, dpi = 500, height = 6, width = 4)
+       plot = tcga_vae_loss_gg, dpi = 500, height = 6, width = 4)
 ggsave(file.path(base_dir, paste0("vae_training_reconstruction_", dataset, ".png")),
-       plot = g, height = 6, width = 4)
+       plot = tcga_vae_loss_gg, height = 6, width = 4)
 
-g
+tcga_vae_loss_gg
 
 # Subset to iterations that may have converged
 tcga_recon_cost_df <- tcga_recon_cost_df %>% dplyr::filter(reconstruction_cost < 4000)
 
-g <- plot_reconstruction_loss(tcga_recon_cost_df)
+tcga_recon_filter_gg <- plot_reconstruction_loss(tcga_recon_cost_df)
 
 ggsave(file.path(base_dir, paste0("reconstruction_cost_subset_converge_", dataset, ".pdf")),
-       plot = g, dpi = 500, height = 3, width = 9)
+       plot = tcga_recon_filter_gg, dpi = 500, height = 3, width = 9)
 ggsave(file.path(base_dir, paste0("reconstruction_cost_subset_converge_", dataset, ".png")),
-       plot = g, height = 3, width = 9)
+       plot = tcga_recon_filter_gg, height = 3, width = 9)
 
-g
+tcga_recon_filter_gg
 
 # Subset to testing non-shuffled data
 tcga_recon_cost_df <- tcga_recon_cost_df %>%
     dplyr::filter(data_type == 'testing', shuffled == 'False')
 
-g <- plot_reconstruction_loss(tcga_recon_cost_df)
+tcga_recon_filter_test_gg <- plot_reconstruction_loss(tcga_recon_cost_df)
```

Comment:
Don't know how easy it would be to do this in R, but maybe want to make a plot function since you're duplicating here

Reply:
`plot_reconstruction_loss` is a custom function. See [here](https://github.com/greenelab/BioBombe/blob/master/4.analyze-components/scripts/util.R#L134)